store/iavl/store.go

```diff
@@ -275,31 +277,38 @@ func (st *Store) Query(req abci.RequestQuery) (res abci.ResponseQuery) {
 			break
 		}
 
-		if req.Prove {
-			value, proof, err := tree.GetVersionedWithProof(key, res.Height)
+		_, res.Value = tree.GetVersioned(key, res.Height)
+		if !req.Prove {
+			break
+		}
+		// Continue to prove existence/absence of value
+		var commitmentProof *ics23.CommitmentProof
+		var err error
+
+		// Must convert store.Tree to iavl.MutableTree with given version to use in CreateProof
```

Comment:
Can you take a look at ics23-iavl to see if I can change this requirement (and accept iavl.ImmutableTree instead). I was just gettng it to work, but in this use case referring to an immutable tree seems better.

Reply:
Yea since you aren't using any Versioned function, you should be able to just swap out `MutableTree` for `ImmutableTree` in the function signatures and everything else should still work.

This will make the code here cleaner